crates/oxc_transformer/src/typescript/enum.rs

```diff
@@ -202,21 +264,12 @@ impl<'a> TypeScript<'a> {
                 );
             }
 
+            prev_member_name = Some(member_name.clone());
             statements.push(self.ctx.ast.expression_statement(member.span, expr));
-
-            // 1 + Foo["x"]
-            default_init = {
-                let one = self.ctx.ast.literal_number_expression(NumericLiteral {
-                    span: SPAN,
-                    value: 1.0,
-                    raw: "1",
-                    base: NumberBase::Decimal,
-                });
-
-                self.ctx.ast.binary_expression(SPAN, one, BinaryOperator::Addition, self_ref)
-            };
         }
 
+        self.enums.insert(enum_name.clone(), previous_enum_members.clone());
```

Comment:
For the following expression:
```TypeScript
enum Foo {
    A = 1,
}
```
We need to store not only the correspondence of A -> 1, but also the correspondence of 1 -> A.
This way, `Foo[1]` can return `A`.